scio-core/src/main/scala/com/spotify/scio/io/FileStorage.scala

```diff
@@ -43,7 +44,7 @@ private[scio] object FileStorage {
 
 private[scio] final class FileStorage(protected[scio] val path: String) {
   private def listFiles: Seq[Metadata] =
-    FileSystems.`match`(path).metadata().asScala
+    FileSystems.`match`(path, EmptyMatchTreatment.DISALLOW).metadata().asScala
```

Comment:
I added this because the behavior of `match` for missing files is inconsistent across file system implementations and some of them will just return a status of `NOT_OK` and an empty list... so it's kind of a breaking change but the only way a user would encounter it is by explicitly creating something like a `TextTap("non/existent/path")`.

Reply:
And in that case it should fail instead of silently returning `Nil` right? :+1: